### PR TITLE
ts-719 implement test to check that server invalidates producer's open transactions when producer's token expires

### PR DESCRIPTION
**Pull request by tomskikh**
That test will be failing until issue #711 will be solved.
Add builder for TestStorageServer.

### CODE REVIEW

File: t-streams/src/test/scala/com/bwsw/tstreams/agents/integration/ServerInvalidatesOpenTransactionWhenProducerTokenExpires.scala [hunk 1]
```diff
@@ -0,0 +1,142 @@
...
+      createTransaction1Latch.countDown()
+      transaction1OpenedLatch.await(awaitTimeout, TimeUnit.SECONDS)
+
+      val transaction2 = producer2.newTransaction(NewProducerTransactionPolicy.ErrorIfOpened)
+      transaction2.send("data")
+      producer2.checkpoint()
+
+      producer1ThreadGroup.interrupt()
+
+      subscriberLatch.await(tokenTtlSec * 2, TimeUnit.SECONDS) shouldBe true
```
Comment (placccebo) [inline t-streams/src/test/scala/com/bwsw/tstreams/agents/integration/ServerInvalidatesOpenTransactionWhenProducerTokenExpires.scala]: should we add the explicit check that ts6 < ts7?

File: t-streams/src/test/scala/com/bwsw/tstreams/agents/integration/ServerInvalidatesOpenTransactionWhenProducerTokenExpires.scala [hunk 2]
```diff
@@ -0,0 +1,142 @@
...
+    factory.setProperty(ConfigurationOptions.Producer.Transaction.ttlMs, 60 * 1000)
+    factory.setProperty(ConfigurationOptions.Producer.Transaction.keepAliveMs, 10 * 1000)
+  }
+
+  override protected def afterAll(): Unit = {
+    TestStorageServer.dispose(server)
+  }
+
+
+  "Subscriber" should "get second transaction if first transaction's producer is broken and it's token is expired " +
```
[the remark this answers is not in the document]
Comment (tomskikh) [reply t-streams/src/test/scala/com/bwsw/tstreams/agents/integration/ServerInvalidatesOpenTransactionWhenProducerTokenExpires.scala]: Fixed

File: t-streams/src/test/scala/com/bwsw/tstreams/agents/integration/ServerInvalidatesOpenTransactionWhenProducerTokenExpires.scala (hunk 1 above)
Comment (tomskikh) [reply t-streams/src/test/scala/com/bwsw/tstreams/agents/integration/ServerInvalidatesOpenTransactionWhenProducerTokenExpires.scala]: Yes. Added.

File: t-streams/src/test/scala/com/bwsw/tstreams/agents/integration/ServerInvalidatesOpenTransactionWhenProducerTokenExpires.scala [hunk 3]
```diff
@@ -0,0 +1,148 @@
...
+    new Thread(
+      firstProducerThreadGroup,
+      () => {
+        val producer1 = factory.getProducer("test_producer_1", partitions)
+        createFirstTransactionLatch.await()
+        val transaction1 = producer1.newTransaction(NewProducerTransactionPolicy.ErrorIfOpened) // ts1
+
+        transaction1.send("data")
+        firstTransactionOpenedLatch.countDown()
+        Thread.sleep(60 * 1000)
```
Comment (bwsw) [inline t-streams/src/test/scala/com/bwsw/tstreams/agents/integration/ServerInvalidatesOpenTransactionWhenProducerTokenExpires.scala]: Remove magic var from here. All awaitings should be declared in one place and (if possible) logically computed from basic values.

File: t-streams/src/test/scala/com/bwsw/tstreams/agents/integration/ServerInvalidatesOpenTransactionWhenProducerTokenExpires.scala [hunk 4]
```diff
@@ -0,0 +1,148 @@
...
+
+
+  "Subscriber" should "get second transaction if first transaction's producer is broken and its token is expired " +
+    "but first transaction's ttl isn't expired" in {
+
+    val firstProducerThreadGroup = new ThreadGroup("producer_1_thread_group")
+
+    val createFirstTransactionLatch = new CountDownLatch(1)
+    val firstTransactionOpenedLatch = new CountDownLatch(1)
+    new Thread(
```
Comment (bwsw) [inline t-streams/src/test/scala/com/bwsw/tstreams/agents/integration/ServerInvalidatesOpenTransactionWhenProducerTokenExpires.scala]: Thread should be named and, please, call t.join() in the end to clean-up the resources. Current implementation without join leads to the lack of them.

Comment (tomskikh) [reply t-streams/src/test/scala/com/bwsw/tstreams/agents/integration/ServerInvalidatesOpenTransactionWhenProducerTokenExpires.scala]: Fixed

File: t-streams/src/test/scala/com/bwsw/tstreams/agents/integration/ServerInvalidatesOpenTransactionWhenProducerTokenExpires.scala (hunk 3 above)
Comment (tomskikh) [reply t-streams/src/test/scala/com/bwsw/tstreams/agents/integration/ServerInvalidatesOpenTransactionWhenProducerTokenExpires.scala]: Fixed

**Comment by tomskikh** [comment]
Will be merged after #747

**Pull request merged**